### PR TITLE
add PermuteChannels transform

### LINKED ISSUE #7620: ChannelShuffle transforms
### 🚀 The feature

Is there any chance torchvision could implement a ChannelShuffle transform?

### Motivation, pitch

Albumenations has it.
https://albumentations.ai/docs/api_reference/augmentations/transforms/#albumentations.augmentations.transforms.ChannelShuffle

### Alternatives

I'm currently implementing this myself by hand or using Albumenations.

### Additional context

Channel Shuffle is useful since it allows me to mix up the input to prevent overfitting.

cc @vfdev-5

### PR DESCRIPTION
**Pull request by pmeier**
Closes #7620. I need to add proper test for the functionals. Happy to bikeshed on the name. I went with `permute_channels`, because that is what we had before. But "shuffle channels" as proposed in #7620 is also an option.

cc @vfdev-5

### CODE REVIEW

File: torchvision/transforms/v2/functional/_color.py
Comment (vadimkantorov) [inline torchvision/transforms/v2/functional/_color.py:688]: should this just be `image = image[..., permutation, :, :]`? maybe this could simplify the code

Comment (vadimkantorov) [reply torchvision/transforms/v2/functional/_color.py:688]: Are JIT considerations still relevant? From my discussions in core, I understood that it is now deprecated, and that Dynamo is the relevant technology. Do you mean that it is also not supported by Dynamo?

Comment (pmeier) [reply torchvision/transforms/v2/functional/_color.py:688]: > From my discussions in core, I understood that it is now deprecated

That would be nice, but so far it is not true. torchscript has not been officially deprecated yet. Hence we still need to keep support for it.

So far I have only seen chatter from individual maintainers that said as much, but my current understanding is that dynamo is not yet a full replacement. Meaning, until that is the case, we can't drop torchscript support.

As soon as there is an official deprecation notice, we can also act on that. Depending on the status of v2, we might go through a deprecation cycle or drop it sooner.

**Review by NicolasHug — COMMENTED** [review]
I'd rather just call it `ChannelShuffle` for consistency with the other package

File: torchvision/transforms/v2/_color.py
Comment (NicolasHug) [inline torchvision/transforms/v2/_color.py:190]: why does this need to be a list? Wasn't it was kept as a tensor in the previous implem?

File: test/test_transforms_v2_refactored.py
Comment (NicolasHug) [inline test/test_transforms_v2_refactored.py:2288]: Please get rid of that and hard-code its actual value where it's used. It's impossible to know what's being permuted otherwise.

That also means we can get rid of `_CORRECTNESS_PERMUTATIONS`.

File: torchvision/transforms/v2/_color.py
Comment (NicolasHug) [inline torchvision/transforms/v2/_color.py:180]: Needs docs

File: torchvision/transforms/v2/functional/_color.py
Comment (NicolasHug) [inline torchvision/transforms/v2/functional/_color.py:649]: Needs docs, in particular for the `permutation` param

File: test/test_transforms_v2_refactored.py
Comment (NicolasHug) [inline test/test_transforms_v2_refactored.py:2337]: This is testing that the indexing semantics are the same in `torch` and in `numpy`, but it doesn't test nor illustrate that the transform is working as expected. I mean, we can keep this test, but it would be good to also assert something like

```
for input_c in range(len(permutation)):
    assert_close(image[input_c, :, :], output[permutation[input_c], :, :]
```

File: test/test_transforms_v2_refactored.py
Comment (NicolasHug) [inline test/test_transforms_v2_refactored.py:2334]: can we also check batched images?

**Review by NicolasHug — APPROVED** [review]
Thanks @pmeier , approving to unblock with a few comments

> I'd rather just call it ChannelShuffle for consistency with the other package

I'm taking this back because of what it would mean for the functional name... All options seem to have some rather annoying downside. I'll let you decide.

File: torchvision/transforms/v2/_color.py
Comment (pmeier) [reply torchvision/transforms/v2/_color.py:190]: I've checked, and we can indeed leave it as tensor even with JIT. The reason I have used a list here is that we have very few functionals that take a tensor as parameter. Since we are never going to script the transform here, we could also have the annotation on the dispatcher as `List[int]`, but pass a tensor from the transform. No strong opinion. Up to you.

Comment (NicolasHug) [reply torchvision/transforms/v2/_color.py:190]: I don't really mind what the annotation is, I just want to make sure we still support tensors so that users don't have to call tolist all the time

File: torchvision/transforms/v2/functional/_color.py
Comment (pmeier) [reply torchvision/transforms/v2/functional/_color.py:649]: I've added a docstring, but so far the v2 functionals are not part of our HTML documentation. LMK how you want me to include this.

File: test/test_transforms_v2_refactored.py
Comment (pmeier) [reply test/test_transforms_v2_refactored.py:2338]: I've refactored the reference function to not rely on indexing, but rather "manually picking" the channels. LMK if that is sufficient.

File: torchvision/transforms/v2/functional/_color.py
Comment (NicolasHug) [reply torchvision/transforms/v2/functional/_color.py:649]: Thanks for the docstring and sorry I wasn't clear enough. The key thing to document (and that is still missing) is what the `permutation` parameter really is. In particular it's not clear whether 

```permutation=[2, ...]```

means "put index 0 at position 2" or "put index 2 at position 0". Right now, only the tests allow to understand that.

**Pull request merged**
